src/spartan/polys/eq.rs

```diff
@@ -55,11 +63,23 @@ impl<Scalar: PrimeField> EqPolynomial<Scalar> {
       evals_left
         .par_iter_mut()
         .zip(evals_right.par_iter_mut())
-        .for_each(|(x, y)| {
-          *y = *x * r;
-          *x -= &*y;
+        .for_each(|(running_product, term)| {
+          // Most-significant bits' evals come first. Each new eval is a product of the previous, and this bit's term --
+          // which is the evaluation, at 0 or 1, of (e_i * x_i + (1 - e_i) * (1 - x_i)).
+          // Substituting r for e_i: (r * x_i + (1 - r) * (1 - x_i))
+          // x_i is 'this bit', which is either 0 or 1.
+          //
```

Comment:
This might refer more directly to Lagrange basis polynomials (PAZK circa Lemma 3.6)

Reply:
I agree with Francois that it might be worth replacing this explanation with a reference to the calculation using Lagrange polynomials (as explained in [`polys/multilinear.rs`](https://github.com/lurk-lab/arecibo/blob/4aa21762a112d47d44d2bda8017e221a38b67cf0/src/spartan/polys/multilinear.rs#L29))

Reply:
I'm open to reworking, but I'm not sure why that's most relevant here. I'm just inlining the Eq polynomial evaluation formula as given in this module's doc. Please feel free to just write something that improves on this, if it's easy for you. The above is just meant to be a literal unpacking of the code to understand how it does what it claims. [NOTE: I'm disclaiming the background knowledge to write an optimally pedantic explanation and inviting anyone who possesses it do so. What I wrote explains what was missing for *me*.]